### PR TITLE
enable tree-shaking and version bumps:

### PR DESCRIPTION
**Pull request by sdalonzo**
- pcln-autocomplete v2.1.0
- pcln-design-system v2.7.0
- pcln-icons v2.3.0
- pcln-modal v2.2.0
- pcln-popover 2.1.0
- pcln-slider v2.3.0

### CODE REVIEW

File: packages/autocomplete/package.json [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-  "main": "dist/index.cjs.js",
-  "module": "dist/index.esm.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "sideEffects": false,
   "scripts": {
-    "prepare": "rollup -c",
-    "clean": "rm -rf node_modules dist",
+    "prepare": "npm run build",
+    "build": "npm run build:cjs && npm run build:esm && npm run build:es",
```
Comment (unscsprt) [inline packages/autocomplete/package.json]: Can we use [npm-run-all](https://www.npmjs.com/package/npm-run-all)? Then it'd just be `npm-run-all build:*` and parallelized.

File: packages/core/package.json [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-  "main": "dist/index.cjs.js",
-  "module": "dist/index.esm.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "sideEffects": false,
   "scripts": {
-    "prepare": "rollup -c",
-    "clean": "rm -rf node_modules dist",
+    "prepare": "npm run clean && npm run build",
+    "build": "npm run build:cjs && npm run build:esm && npm run build:es",
```
Comment (unscsprt) [inline packages/core/package.json]: See https://github.com/priceline/design-system/pull/643/files#r344892544

File: packages/icons/package.json [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
+  "sideEffects": false,
   "scripts": {
     "prepare": "npm run clean && npm run build:svg && npm run build:index && npm run build:copy && npm run build:babel",
-    "clean": "rm -rf node_modules components lib",
+    "clean": "rimraf components lib",
     "build:svg": "svgr svg -d components",
     "build:index": "node ./create-index.js",
     "build:copy": "cp src/Icon.js src/Svg.js src/index.js components",
-    "build:babel": "babel components -d lib",
+    "build:babel": "npm run build:cjs && npm run build:esm && npm run build:es && npm run build:esm-lib",
```
Comment (unscsprt) [inline packages/icons/package.json]: See https://github.com/priceline/design-system/pull/643/files#r344892544

File: packages/popover/package.json [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
-  "main": "dist/index.cjs.js",
-  "module": "dist/index.esm.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "sideEffects": false,
   "scripts": {
-    "prepare": "rollup -c",
-    "clean": "rm -rf node_modules dist",
+    "prepare": "npm run clean && npm run build",
+    "build": "npm run build:cjs && npm run build:esm && npm run build:es",
```
Comment (unscsprt) [inline packages/popover/package.json]: See https://github.com/priceline/design-system/pull/643/files#r344892544

File: packages/core/package.json [hunk 5]
```diff
@@ -1,14 +1,20 @@
 {
   "name": "pcln-design-system",
-  "version": "2.6.2",
+  "version": "2.7.0",
   "description": "Priceline Design System",
-  "main": "dist/index.cjs.js",
-  "module": "dist/index.esm.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "sideEffects": false,
```
Comment (jes708) [inline packages/core/package.json]: I think apps that have a server-side implementation of react-loadable may run into an issue with this.

Comment (jes708) [reply packages/core/package.json]: Perhaps you could make a prerelease version so I can confirm if there are breaking changes.

File: packages/autocomplete/package.json (hunk 1 above)
Comment (James300) [reply packages/autocomplete/package.json]: TIL. This is really cool

File: packages/core/src/Icon.js [hunk 6]
```diff
@@ -1,3 +1,3 @@
-import Icon from 'pcln-icons'
+import Icon from 'pcln-icons/dist/esm/Icon'
```
Comment (James300) [inline packages/core/src/Icon.js]: Can we delete this component now that we are going to v3?

**Comment by dansternfeld1** [comment]
This is declined in favor of #644

**Pull request closed without merging**